src/libraries/System.Private.CoreLib/src/System/Globalization/CalendarData.iOS.cs

```diff
@@ -0,0 +1,111 @@
+                // note that when formatting non-leap year dates, 7th month shouldn't get used at all.
+                saMonthNames[5] = saMonthNames[6];
+                saMonthNames[6] = leapHebrewMonthName;
+
+            }
+            this.saAbbrevMonthNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevMonthNames).Split("||"), calendarId, ref leapHebrewMonthName);
+            this.saMonthGenitiveNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.MonthGenitiveNames).Split("||"), calendarId, ref leapHebrewMonthName);
+            this.saAbbrevMonthGenitiveNames = NormalizeMonthArray(GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevMonthGenitiveNames).Split("||"), calendarId, ref leapHebrewMonthName);
+
+            this.saEraNames = NormalizeEraNames(calendarId, GetCalendarInfoNative(localeName, calendarId, CalendarDataType.EraNames).Split("||"));
+            this.saAbbrevEraNames = Array.Empty<string>();//NormalizeEraNames(calendarId, GetCalendarInfoNative(localeName, calendarId, CalendarDataType.AbbrevEraNames).Split("||"));
+
+            return this.sNativeName != null && this.saShortDates != null && this.saLongDates != null && this.saYearMonths != null &&
+                   this.saDayNames != null && this.saAbbrevDayNames != null && this.saSuperShortDayNames != null && this.saMonthNames != null &&
+                   this.saAbbrevMonthNames != null && this.saMonthGenitiveNames != null && this.saAbbrevMonthGenitiveNames != null &&
+                   this.saEraNames != null && this.saAbbrevEraNames != null;
+        }
+
+        private static string[] NormalizeEraNames(CalendarId calendarId, string[]? eraNames)
+        {
+            // .NET expects that only the Japanese calendars have more than 1 era.
+            // So for other calendars, only return the latest era.
+            if (calendarId != CalendarId.JAPAN && calendarId != CalendarId.JAPANESELUNISOLAR && eraNames?.Length > 0)
+            {
+                string[] latestEraName = new string[] { eraNames![eraNames.Length - 1] };
+                eraNames = latestEraName;
+            }
+
+            return eraNames ?? Array.Empty<string>();
+        }
+
+        private static string[] NormalizeMonthArray(string[] months, CalendarId calendarId, ref string? leapHebrewMonthName)
+        {
+            if (months.Length == 13)
+                return months;
+
+            string[] normalizedMonths = new string[13];
+            // the month-name arrays are expected to have 13 elements.  If only returns 12, add an
+            // extra empty string to fill the array.
+            if (months.Length == 12)
+            {
+                normalizedMonths[12] = "";
+                months.CopyTo(normalizedMonths, 0);
+                return normalizedMonths;
+            }
```

Comment:
And I think we should allocate only after the length == 13 or 14 logic, so bump this down.

Reply:
We need to allocate also when `length == 14`. We should always return 13 length array